odl/solvers/functional/default_functionals.py

```diff
@@ -0,0 +1,683 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Default functionals defined on any (reasonable) space."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+import numpy as np
+
+from odl.solvers.functional.functional import Functional
+from odl.operator.operator import Operator
+from odl.solvers.advanced.proximal_operators import (proximal_l1,
+                                                     proximal_cconj_l1,
+                                                     proximal_l2,
+                                                     proximal_cconj_l2,
+                                                     proximal_l2_squared,
+                                                     proximal_cconj_l2_squared)
+
+from odl import (ZeroOperator, IdentityOperator)
+
+
+__all__ = ('L1Norm', 'L2Norm', 'L2NormSquare', 'ZeroFunctional',
+           'ConstantFunctional')
+
+
+# TODO: Implement some of the missing gradients
+
+class L1Norm(Functional):
+    """The functional corresponding to L1-norm."""
+
+    def __init__(self, domain):
+        """Initalizes an instance of the L1-norm functional.
+
+        Parameters
+        ----------
+        domain : `LinearSpace`
+            Set of elements on which the functional can be evaluat
+        """
+        super().__init__(domain=domain, linear=False, convex=True,
+                         concave=False, smooth=False, grad_lipschitz=np.inf)
+
+    def _call(self, x):
+        """Applies the functional to the given point.
+
+        Parameters
+        ----------
+        x : `LinearSpaceVector`
+            Element in the domain of the functional.
+
+        Returns
+        -------
+        `self(x)` : `element` in the `field` of the ``domain``.
+            Evaluation of the functional.
+        """
+        return np.abs(x).inner(self.domain.one())
+
+    @property
+    def gradient(x):
+        """Gradient operator of the L1-functional."""
+        # It does not exist on all of the space
+        raise NotImplementedError
+
+    def proximal(self, sigma=1.0):
+        """Return the proximal operator of the L1-functional.
+
+        Parameters
+        ----------
+        sigma : positive float, optional
+            Regularization parameter of the proximal operator
+
+        Returns
+        -------
+        out : Operator
+            Domain and range equal to domain of functional
+        """
+
+        return proximal_l1(space=self.domain)(sigma)
+
+    @property
+    def conjugate_functional(self):
+        """The convex conjugate functional of the L1-norm."""
+        functional = self
+
+        class L1ConjugateFunctional(Functional):
```

Comment:
Preferably they should (if possible)  be called something about what they do (`IndicatorLinftyUnitBall`, or something) instead of only being known as the dual of something.